File: backend/src/app/schemas/product.py
from datetime import datetime

from pydantic import BaseModel


class BaseProductDTO(BaseModel):
    sku: int
    name: str
    manufactor: str
    product_measure: str
    product_amount: float
    product_volume: float
    manufacture_date: datetime
    expiry_date: datetime


class ProductOutDTO(BaseProductDTO):
    id: int


class ProductCreateDTO(BaseProductDTO):
    pass

File: frontend/src/lib/data/product.ts
import { fetchJSON } from "$lib/fetch";

export type Product = {
    id: number;
    sku: number;
    name: string;
    manufactor: string;
    product_measure: string;
    product_amount: number;
    product_volume: number;
    manufacture_date: string;
    expiry_date: string;
    category: string;
};

export async function fetchProducts(): Promise<Product[]> {
    return fetchJSON<Product[]>("/products").then(x => x.data);
}
